photon-ml/src/main/scala/com/linkedin/photon/ml/function/ValueAndGradientAggregator.scala

```diff
@@ -15,24 +15,21 @@
 package com.linkedin.photon.ml.function
 
 import breeze.linalg.{DenseVector, Vector, axpy}
-import com.linkedin.photon.ml.data.{LabeledPoint, ObjectProvider}
-import com.linkedin.photon.ml.normalization.NormalizationContext
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.rdd.RDD
 
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.function.glm.PointwiseLossFunction
+import com.linkedin.photon.ml.normalization.NormalizationContext
 
+// TODO: Better document this algorithm, especially normalization.
```

Comment:
@TDDFT I know the one. It's very thorough, but like you said I'm not sure how to reconcile it as it's an internal document. I'm going to make it a separate issue for now.